src/sentry/static/sentry/app/components/events/interfaces/spans/filter.tsx

```diff
@@ -0,0 +1,342 @@
+                    />
+                    <OperationName>{operationName}</OperationName>
+                    <CheckboxFancy
+                      isChecked={isActive}
+                      onClick={event => {
+                        event.stopPropagation();
+                        this.props.toggleOperationNameFilter(operationName);
+                      }}
+                    />
+                  </ListItem>
+                );
+              })}
+            </List>
+          </MenuContent>
+        </DropdownControl>
+      </Wrapper>
+    );
+  }
+}
+
+const FilterLabel = styled('span')`
+  margin-left: ${space(1)};
+`;
+
+const Wrapper = styled('div')`
+  position: relative;
+  display: flex;
+
+  margin-right: ${space(1)};
+`;
+
+const StyledDropdownButton = styled(DropdownButton)<{hasDarkBorderBottomColor?: boolean}>`
+  white-space: nowrap;
+  max-width: 200px;
+
+  z-index: ${p => p.theme.zIndex.dropdown};
+
+  &:hover,
+  &:active {
+    ${p =>
+      !p.isOpen &&
+      p.hasDarkBorderBottomColor &&
+      `
+          border-bottom-color: ${p.theme.button.primary.border};
+        `}
+  }
+
+  ${p =>
+    !p.isOpen &&
+    p.hasDarkBorderBottomColor &&
+    `
+      border-bottom-color: ${p.theme.button.primary.border};
+    `}
+`;
+
+const MenuContent = styled('div')`
+  max-height: 250px;
+  overflow-y: auto;
+  border-top: 1px solid ${p => p.theme.gray400};
+`;
+
+const Header = styled('div')`
+  display: grid;
+  grid-template-columns: auto min-content;
+  grid-column-gap: ${space(1)};
+  align-items: center;
+
+  margin: 0;
+  background-color: ${p => p.theme.gray100};
+  color: ${p => p.theme.gray500};
+  font-weight: normal;
+  font-size: ${p => p.theme.fontSizeMedium};
+  padding: ${space(1)} ${space(2)};
+  border-bottom: 1px solid ${p => p.theme.borderDark};
+`;
+
+const List = styled('ul')`
+  list-style: none;
+  margin: 0;
+  padding: 0;
+`;
+
+const ListItem = styled('li')<{isChecked?: boolean}>`
+  display: grid;
+  grid-template-columns: max-content 1fr max-content;
+  grid-column-gap: ${space(1)};
+  align-items: center;
+  padding: ${space(1)} ${space(2)};
+  border-bottom: 1px solid ${p => p.theme.borderDark};
+  :hover {
+    background-color: ${p => p.theme.gray100};
+  }
+  ${CheckboxFancy} {
+    opacity: ${p => (p.isChecked ? 1 : 0.3)};
+  }
+
+  &:hover ${CheckboxFancy} {
+    opacity: 1;
+  }
+
+  &:hover span {
+    color: ${p => p.theme.blue400};
+    text-decoration: underline;
+  }
+`;
+
+const OperationDot = styled('div')`
+  content: '';
+  display: block;
+  width: 8px;
+  min-width: 8px;
+  height: 8px;
+  margin-right: ${space(1)};
+  border-radius: 100%;
+`;
+
+const OperationName = styled('div')`
+  font-size: ${p => p.theme.fontSizeMedium};
+  text-overflow: ellipsis;
+  white-space: nowrap;
+  overflow: hidden;
+`;
+
+export function toggleFilter(
+  previousState: ActiveOperationFilter,
+  operationName: string
+): ActiveOperationFilter {
+  if (previousState.type === 'no_filter') {
```

Comment:
wasn't it better if the types were an ENUM?

Reply:
it's simpler if it's a string. This is a discriminated union: https://basarat.gitbook.io/typescript/type-system/discriminated-unions#discriminated-union